### ISSUE TITLE
Too may open files error when loading 50+ images

### ISSUE THREAD

**Issue by timkoers** [square/picasso#1082]
I'm using Picasso to load images to ImageViews, and I worked great using a RecyclerView. Because the RecyclerView messed up the list (images where being loaded from an other item (so the item 4 is being displayed, and the image from item 7 is loaded in the displayed item), so I switched to the ListView)

But now, the views that aren't on the screen don't get destroyed and Picasso must load them all. This works without any performance inpacts, but when I scroll quickly to the bottom and back up several times, my app stops working and I get the following logcat error:

http://pastebin.com/nXpTeDXG

And I get the ANR.

The code that loads the image is:

http://pastebin.com/UBusgyQk

And the code that us used to detect scrolls (and stop Picasso) is:

http://pastebin.com/ym39zCDd

Btw, images are loading (and visible) so that part is working.

I've searched on Google to check for solutions about this error, but I couldn't find anything useful.

I'm using Picasso 2.5.2 and OkHTTP 2.4.0.

What is going on here? 

Thanks for the solution/fix!

Tim

**Comment by JakeWharton** [comment]
Do not create a new Picasso instance every time you want to load an image. Create it once and re-use it everywhere.

**Comment by timkoers** [comment]
I've created the an picasso instance first, and reused it everytime I want to load an image.

However, the error is still happening. When I took a look at the ADM, and enabled the thread updating, I found out that Picasso creates alot (but I mean alot) of threads. It looks like 3 threads per image. This isn't how it's supposed to be. I started using an ExecutorService with Picasso, and it's still happening.
